From 691e3a443996f783fd91f686568d93f1b96bad46 Mon Sep 17 00:00:00 2001
From: Pavel Grunt <pgrunt@redhat.com>
Date: Mon, 2 Feb 2015 17:25:52 +0100
Subject: [PATCH] Do not add "https://" and "/api" to oVirt URI

It is deprecated since govirt 0.3.1 (and virt-viewer already depends on
govirt 0.3.2).

Silences:
(remote-viewer:19420): libgovirt-WARNING **: Passing a full http:// or https:// URI to ovirt_proxy_new() is deprecated
(remote-viewer:19420): libgovirt-WARNING **: Passing an URI ending in /api to ovirt_proxy_new() is deprecated

(cherry picked from commit 45c6fc9b06eab5b51243d4e46e7fd08fb94766c3)
---
 src/remote-viewer.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/remote-viewer.c b/src/remote-viewer.c
index 5c17b22..2a6ad6d 100644
--- a/src/remote-viewer.c
+++ b/src/remote-viewer.c
@@ -668,7 +668,7 @@ parse_ovirt_uri(const gchar *uri_str, char **rest_uri, char **name, char **usern
 
     if (uri->path == NULL) {
         *name = NULL;
-        *rest_uri = g_strdup_printf("https://%s/api/", uri->server);
+        *rest_uri = g_strdup(uri->server);
         xmlFreeURI(uri);
         return TRUE;
     }
@@ -691,8 +691,7 @@ parse_ovirt_uri(const gchar *uri_str, char **rest_uri, char **name, char **usern
 
     /* build final URI */
     rel_path = g_strjoinv("/", path_elements);
-    /* FIXME: how to decide between http and https? */
-    *rest_uri = g_strdup_printf("https://%s%s/api/", uri->server, rel_path);
+    *rest_uri = g_strdup_printf("%s%s", uri->server, rel_path);
     *name = vm_name;
     g_free(rel_path);
     g_strfreev(path_elements);
